From: Richard Leitner <richard.leitner@skidata.com>
Date: Thu, 23 May 2019 13:54:49 +0200
Subject: [PATCH] rtc: s35390a: set uie_unsupported
Origin: https://git.kernel.org/linus/c0e12848be091e8410fb427f080f2e0149123443
Bug-Debian: https://bugs.debian.org/932845

Alarms are only supported on a per minute basis. This is why
uie_unsupported is set. Furthermore issue a warning when a second based
alarm is requested.

Signed-off-by: Richard Leitner <richard.leitner@skidata.com>
Signed-off-by: Alexandre Belloni <alexandre.belloni@bootlin.com>
---
 drivers/rtc/rtc-s35390a.c | 6 ++++++
 1 file changed, 6 insertions(+)

Index: linux/drivers/rtc/rtc-s35390a.c
===================================================================
--- linux.orig/drivers/rtc/rtc-s35390a.c
+++ linux/drivers/rtc/rtc-s35390a.c
@@ -288,6 +288,9 @@ static int s35390a_rtc_set_alarm(struct
 		alm->time.tm_min, alm->time.tm_hour, alm->time.tm_mday,
 		alm->time.tm_mon, alm->time.tm_year, alm->time.tm_wday);
 
+	if (alm->time.tm_sec != 0)
+		dev_warn(&client->dev, "Alarms are only supported on a per minute basis!\n");
+
 	/* disable interrupt (which deasserts the irq line) */
 	err = s35390a_set_reg(s35390a, S35390A_CMD_STATUS2, &sts, sizeof(sts));
 	if (err < 0)
@@ -502,6 +505,9 @@ static int s35390a_probe(struct i2c_clie
 		goto exit_dummy;
 	}
 
+	/* supports per-minute alarms only, therefore set uie_unsupported */
+	s35390a->rtc->uie_unsupported = 1;
+
 	if (status1 & S35390A_FLAG_INT2)
 		rtc_update_irq(s35390a->rtc, 1, RTC_AF);
 
